netsuite/11.3/modules/ROOT/pages/netsuite-examples.adoc

```diff
@@ -0,0 +1,751 @@
+http://www.mulesoft.org/schema/mule/netsuite http://www.mulesoft.org/schema/mule/netsuite/current/mule-netsuite.xsd
+http://www.mulesoft.org/schema/mule/http http://www.mulesoft.org/schema/mule/http/current/mule-http.xsd
+http://www.mulesoft.org/schema/mule/ee/core http://www.mulesoft.org/schema/mule/ee/core/current/mule-ee.xsd">
+
+
+	<http:listener-config name="HTTP_Listener_config" doc:name="HTTP Listener config">
+		<http:listener-connection host="0.0.0.0" port="8081" />
+	</http:listener-config>
+	<netsuite:config name="tokenConfig" doc:name="NetSuite SOAP Config">
+		<netsuite:token-based-authentication-connection consumerKey="${netsuite.consumerKey}" consumerSecret="${netsuite.consumerSecret}" tokenId="${netsuite.tokenId}" tokenSecret="${netsuite.tokenSecret}" account="${netsuite.account}"/>
+	</netsuite:config>
+
+    <flow name="customer-advanced-search-flow">
+		<http:listener doc:name="Listener" config-ref="HTTP_Listener_config" path="customer" />
+		<ee:transform doc:name="Search Criteria">
+			<ee:message >
+				<ee:set-payload ><![CDATA[%dw 2.0
+output application/xml
+ns ns0 urn:messages_2020_2.platform.webservices.netsuite.com
+ns ns01 urn:common_2020_2.platform.webservices.netsuite.com
+ns ns02 urn:core_2020_2.platform.webservices.netsuite.com
+ns ns03 urn:relationships_2020_2.lists.webservices.netsuite.com
+ns xsi http://www.w3.org/2001/XMLSchema-instance
+---
+{
+	ns0#search: {
+		ns0#searchRecord @("xmlns:ns03": ns03, xsi#"type": "ns03:CustomerSearchAdvanced"): {
+			ns03#criteria: {
+				ns03#basic: {
+					ns01#stage @(operator: "anyOf"): {
+						ns02#searchValue: [attributes.queryParams["stage"]]
+					},
+					ns01#dateCreated @(operator: "after"): {
+						ns02#searchValue: "2015-01-01T00:00:00.000-08:00"
+					}
+				}
+			}
+		}
+	}
+}]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+		<netsuite:search doc:name="Search Customer Advanced" config-ref="tokenConfig" returnSearchColumns="false" key="CustomerSearchAdvanced"/>
+		<foreach doc:name="For Each">
+			<ee:transform doc:name="Transform to JSON">
+			<ee:message>
+				<ee:set-payload><![CDATA[%dw 2.0
+output application/json
+ns ns0 urn:core_2020_2.platform.webservices.netsuite.com
+---
+payload.ns0#record
+]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+			<logger level="INFO" doc:name="Logger" message="#[payload]" />
+		</foreach>
+	</flow>
+</mule>
+----
+
+== Custom Records and Fields
+
+The following examples show how to add, get, and delete custom records with
+custom fields.
+
+[[add-custom-record]]
+=== Add a Custom Record
+
+This example shows how to add a custom record with custom fields using the *Add*
+operation. This example is generated using DataSense.
+
+Pick a recordType in the operation to generate metadata. In this example, the
```

Comment:
I am finding this confusing. I don't see this key in the XML. Am I missing something?

Reply:
Yes, it does have another name, see this line https://github.com/mulesoft/docs-connectors/pull/1528/files/860ed6edf1aa97eae34826845c9a282e9e362ed1#diff-18e4957035af542ae9c6a98da0d7deec9f4518fdd03cf117686d335fbb45b095R682
```suggestion
Pick a type in the operation to generate metadata. In this example, the
```